Improve usage as a library

schollz opened the pull request:
Hi @common-nighthawk , I love this code you wrote! I wanted to use it for a CAPTCHA. I've made some changes and I'm putting them here. It will look like a lot of changes, but its not (for some reason your code wasn't gofmt-ed, so there's a ton of whitespace changes).

First of all, what is the license on your code? If its not open, its fine, I'll follow whatever you decide.

My changes: 

- I made one big change - that is to run `go-bindata fonts/` and then add that to the repo. Now all of the fonts are built-in and you can import any fonts you want even if you use Windows (I use Windows so this was great for me).

- I made a small change to add a `String()` method to the `figure` struct. That means that you can print a figure with just `fmt.Println(myfigure)` and also means that you can create strings from the figures (`myfigure.String()`) which is useful in my case since I want to use these in a captcha.

- I added a test (tests are now easy since you can stringify things).

That's basically all the changes I made. The function `newFontFromReader()` is basically deprecated now, since its no longer needed to support adding random flf files, but I haven't deleted it in this.

I'm going to continue to tweak things a little bit for my CAPTCHA needs, and I'm happy to follow your API if you want PRs. However, if you don't want PRs, no problem, I'll just continue on my own way :)

common-nighthawk commented on font.go:
```diff
@@ -1,89 +1,100 @@
-  font.setLetters(scanner)
-  return font
+	font.setName(name)
+	fontBytes, err := Asset(path.Join("fonts", name+".flf"))
```
This line needs to be `fontBytes, err := Asset(path.Join("fonts", font.name+".flf"))` in order for the default font to work.  That is, using `name` instead of `font.name` undoes the work of the line above.

common-nighthawk commented on figure.go:
```diff
@@ -1,67 +1,58 @@
+			printRow += charRowText
+		}
+		if r < figure.font.baseline || len(strings.TrimSpace(printRow)) > 0 {
+			rows = append(rows, strings.TrimSpace(printRow))
```
:+1:

common-nighthawk commented on figure_test.go:
```diff
@@ -0,0 +1,16 @@
+)
+
+func ExampleAlphabet() {
+	myFigure := NewFigure("Hello world", "alphabet", true)
```
Would you mind adding a test to ensure the default font works?  `myFigure := NewFigure("Hello world", "", true)`.

common-nighthawk commented on figure.go:
```diff
@@ -1,67 +1,58 @@
-  return figure{phrase, font, strict}
-}
-
-func NewFigureWithFont(phrase string, reader io.Reader, strict bool) figure {
```
Can we leave this method?  Details in main comment.

schollz commented:
Hi @common-nighthawk , done.

There's one weird thing though: check out [the test for default](https://github.com/common-nighthawk/go-figure/pull/6/commits/8babb42a02cfe7735e569205375941d19b7d46e0). It's font is missing some left-space on the first line. I have no idea why. I removed the trimspace I put in and replaced with a trimright (https://github.com/common-nighthawk/go-figure/pull/6/commits/0a8e758e79c4756745b4814d8d21f492b7a908d8) but that didn't fix it. I'm not sure whats going on there, maybe you could take a look.

common-nighthawk commented:
Awesome!  I'm glad we added that test!--looks like `TrimSpace` wasn't exactly correct.

I'm not sure what's going on with the test.  But if you edit the output to be what you'd expect (inserting 2 spaces on line 22), the test still passes.  I confirmed everything outputs as expected locally as well.

I'll go ahead and merge.  Thanks again!!
